Match kwargs in SubgrpahMatcher

SherlockNoMad opened the pull request:
Pattern node and target node must have identical kwargs now... 

Use envvar `LOGLEVEL=INFO` to turn on the logging message for easier debugging... 


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85617

SherlockNoMad committed:
Match kwargs in SubgrpahMatcher

[ghstack-poisoned]

SherlockNoMad committed:
Update on "Match kwargs in SubgrpahMatcher"




[ghstack-poisoned]

jerryzh168 commented on torch/fx/passes/utils/matcher_utils.py:
```diff
@@ -172,7 +179,14 @@ def _match_nodes(self, pn: Node, gn: Node, match: InternalMatch) -> bool:
         gn_flatten_args, _ = pytree.tree_flatten(gn.args)
 
-        if len(pn_flatten_args) == len(gn_flatten_args):
+        if pn.kwargs.keys() == gn.kwargs.keys():
```
nit: do we need to make sure the keys are the same as well?

SherlockNoMad replied:
pn.kwargs.keys() == gn.kwargs.keys() 

hmm, is something missing with this check?

jerryzh168 commented on test/fx/test_subgraph_rewriter.py:
```diff
@@ -656,3 +668,99 @@ def replacement(x):
+
+        self.assertTrue(found_repalcement_node)
+
+    def test_subgraph_rewriter_local_revert(self):
```
could you add some description on how this tests the functionality?

SherlockNoMad replied:
Done.

jerryzh168 commented:
looks good, might be good to separate the functionality change and logging change in the future

SherlockNoMad committed:
Update on "Match kwargs in SubgrpahMatcher"

Pattern node and target node must have identical kwargs now... 

Use envvar `LOGLEVEL=INFO` to turn on the logging message for easier debugging... 


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85617

[ghstack-poisoned]

SherlockNoMad commented:
> looks good, might be good to separate the functionality change and logging change in the future

Thanks for reviewing this!! Will try to keep it as separated PR in the future.

SherlockNoMad commented:
@pytorchbot merge -g

SherlockNoMad commented:
@pytorchbot merge -g

davidberard98 reviewed (approved):
Duplicate Jerry's stamp
